test_plans/atomic_ref.asciidoc

```diff
@@ -0,0 +1,480 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for atomic_ref
+
+This is a test plan for the APIs described in SYCL 2020 section 4.15.3. Atomic references
+
+== Testing scope
+
+The test plan covers non-atomic behavior, the latter will be covered in a separate test plan
+
+=== Device coverage
+
+All of the tests described below are performed on the default device that is selected on the CTS command line
+
+== Type coverage
+
+All of the tests described below are performed using each of the
+following types as the `ReferencedType`.
+
+In Regular mode:
+
+* `int`
+* `float`
+
+In full conformance mode:
+
+* `int`
+* `unsigned int`
+* `long`
+* `unsigned long`
+* `float`
+
+In addition, if the device has `aspect::atomic64`, the following types are tested:
+
+* `long long`
+* `unsigned long long`
+* `double`
+
+Additionaly run same tests for `ReferencedType*`.
+
+Note: using T as `ReferencedType` or `ReferencedType*`.
+
+== Tests
+
+All non static tests run on following cases:
```

Comment:
Do we have a definition of "non static test" somewhere?

Reply:
Note that I'm fine with term "[non] static test" if it is documented somewhere. To me, the most appealing description would be something like "runtime checks"/"compile-time checks" because I see all CTS as runtime tests. One more idea I have is to say something like "test cases which check behavior and not interface": interface could be checked with compile-time checks, whilst behavior can only be checked at runtime.